MarcusBarnes/mik
# Work on #461 - Inconsistent default configuration between toolchains

**Github issue**: (#461 )

# What does this Pull Request do?

Makes the default child sequence separator in the CSV compound toolchain consistent with the page sequence separator in the CSV books and newspapers toolchains (i.e., a hypehn).

# What's new?

Changed the default separator in the CsvCompound writer class and input validator class, and renamed test files to use that separator. Also marked the `testCsvCompoundInputValidator()` test as skipped (see below).

# How should this be tested?

Running `phpunit` in the mik directory should should pass, but now, two tests are reported as skipped. As I tried to get `testCsvCompoundInputValidator()`, unrelated tests started failing. There is obviously something unrelated wrong with our tests for the CSV Compound toolchain, a suspicion supported by also skipped `testWritePackages()` test in CsvCompoundToolchainTest.php.

To test the change, unzip the attached zip file and run MIK (after modifying any paths in the .ini file to suit your system). The output should be as expected. To test changing the default separator, swap:

```
input_directory = "issue-461/dash"
; input_directory = "issue-461/underscore"
```
and uncomment

```
; child_sequence_separator = '_'
```

and rerun mik. All output should be as expected.

# Additional Notes

We will need to update https://github.com/MarcusBarnes/mik/wiki/Toolchain:-CSV-compound-objects to indicate the new default, and also add a note about when the default was changed.

# Interested parties

@bondjimbond 

[issue-461.zip](https://github.com/MarcusBarnes/mik/files/1938861/issue-461.zip)
